ldap/servers/slapd/back-ldbm/db-mdb/mdb_config.c

```diff
@@ -0,0 +1,1375 @@
+        } else if (result == LDAP_ALREADY_EXISTS) {
+            slapi_log_err(SLAPI_LOG_TRACE, "dbmdb_ctx_t_add_dse_entries", "Database config entry [%s] already exists - skipping\n", ebuf);
+        } else {
+            slapi_log_err(SLAPI_LOG_ERR, "dbmdb_ctx_t_add_dse_entries",
+                          "Unable to add config entry [%s] to the DSE: %d %d\n",
+                          ebuf, result, rc);
+        }
+        slapi_pblock_destroy(util_pb);
+    }
+
+    return 0;
+}
+
+/* used to add a single entry, special case of above */
+int
+dbmdb_ctx_t_add_dse_entry(struct ldbminfo *li, char *entry, int flags)
+{
+    char *entries[] = {"%s", ""};
+
+    return dbmdb_ctx_t_add_dse_entries(li, entries, entry, NULL, NULL, flags);
+}
+
+
+/*------------------------------------------------------------------------
+ * Get and set functions for mdb variables
+ *----------------------------------------------------------------------*/
+
+static void *
+dbmdb_ctx_t_db_max_size_get(void *arg)
+{
+    struct ldbminfo *li = (struct ldbminfo *)arg;
+    dbmdb_ctx_t *conf = li->li_dblayer_config;
+
+    return  (void *)((uintptr_t)(conf->dsecfg.max_size));
+}
+
+static int
+dbmdb_ctx_t_db_max_size_set(void *arg, void *value, char *errorbuf __attribute__((unused)), int phase __attribute__((unused)), int apply)
+{
+    struct ldbminfo *li = (struct ldbminfo *)arg;
+    dbmdb_ctx_t *conf = li->li_dblayer_config;
+    int retval = LDAP_SUCCESS;
+    uint64_t val = (uint64_t)((uintptr_t)value);
+    uint64_t curval = val;
+
+    if (conf->limits.max_size < conf->limits.min_size) {
+        slapi_log_err(SLAPI_LOG_ERR, "dbmdb_ctx_t_db_max_size_set",
+                "Not enough space on %s home directory to host a database.\n", conf->home);
+        return LDAP_UNWILLING_TO_PERFORM;
+    }
+    if (val != 0) {
+        /* Let check the limits */
+        if (curval < conf->limits.min_size) {
+            curval = conf->limits.min_size;
+        }
+        if (curval > conf->limits.max_size) {
+            curval = conf->limits.max_size;
+        }
+        if (curval > val) {
+            slapi_log_err(SLAPI_LOG_WARNING, "dbmdb_ctx_t_db_max_size_set",
+                "nsslapd-mdb-max-size value is too small."
+                " Increasing the value from %lud to %lud\n", val, curval);
+        } else if (curval < val) {
+            slapi_log_err(SLAPI_LOG_WARNING, "dbmdb_ctx_t_db_max_size_set",
+                "nsslapd-mdb-max-size value is not compatible with current partition size."
+                " Decreasing the value from %lud to %lud\n", val, curval);
+       }
+        val = curval;
+    }
+
+    if (apply) {
+        conf->dsecfg.max_size = val;
+        if (CONFIG_PHASE_RUNNING == phase) {
+            slapi_log_err(SLAPI_LOG_NOTICE, "dbmdb_ctx_t_db_max_size_set",
+                          "New nsslapd-mdb-max-size will not take affect until the server is restarted\n");
```

Comment:
Max size can only be set once at db creation time. Changes to this value will NOT take effect later, the db needs recreation.

Reply:
I think you are wrong as mdb_env_set_mapsize document explictly says that size could be increased by another process
 and there is an error code to handle that. 
That said size yo not be decreased without rebuilding the db.

Reply:
okay, so we need a check to make sure this isn't reduced?

Reply:
I do not think we should prevent that because:
   - mdb_open_env will silently increase the value if it is too small.
   - this will allow to reduce the size if the db get rebuild

Reply:
Okay then :)